s4e-backend/src/main/java/pl/cyfronet/s4e/controller/AppUserController.java

```diff
@@ -142,11 +145,26 @@
             @ApiResponse(code = 200, message = "User was added")
     })
     @PostMapping("/institutions/{institution}/users")
-    public ResponseEntity<?> addUserToInstitution(@RequestBody @Valid CreateUserWithGroupsRequest request ) {
+    public ResponseEntity<?> addUserToInstitution(@RequestBody @Valid CreateUserWithGroupsRequest request) {
 
         return ResponseEntity.ok().build();
     }
 
+    @ApiOperation("Get user profile")
+    @ApiResponses({
+            @ApiResponse(code = 200, message = "User was added"),
+            @ApiResponse(code = 403, message = "User not authorized"),
+            @ApiResponse(code = 404, message = "User wasn't found")
+    })
+    @GetMapping("/users/me")
+    @PreAuthorize("isAuthenticated()")
+    public AppUserResponse getMe() throws NotFoundException {
+        UserDetails userDetails = (UserDetails) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        val optionalUser = appUserService.findByEmail(userDetails.getUsername())
```

Comment:
Also, this is not optional

Reply:
how so? AppUserService::Optional<AppUser> findByEmail(String email)